engine/src/main/java/com/datatorrent/stram/engine/InputNode.java

```diff
@@ -69,24 +69,27 @@ public final void run()
     long spinMillis = context.getValue(OperatorContext.SPIN_MILLIS);
     final boolean handleIdleTime = operator instanceof IdleTimeHandler;
 
-    boolean insideWindow = applicationWindowCount != 0;
+    boolean insideApplicationWindow = applicationWindowCount != 0;
     boolean doCheckpoint = false;
+    boolean insideStreamingWindow = false;
 
     try {
       while (alive) {
         Tuple t = controlTuples.sweep();
         if (t == null) {
-          if (insideWindow) {
+          if (insideApplicationWindow) {
```

Comment:
This should be insideStreamingWindow as well. For example handleIdleTime should not be called between streaming window boundaries.